### PR TITLE
updated gitignore

### PR DESCRIPTION
**Pull request by mister-abdurahman**

### CODE REVIEW

**Comment by xquanluu** [comment]
Screenshot 2024-08-26 at 14 29 17

From figma design, you can see that:
1/ Clickable area is the small one inside the borrder radius rectangle, not the whole row.
2/ We don't have the devider.

**Comment by xquanluu** [comment]
Screenshot 2024-08-26 at 14 33 49

From figma:
1/ show button is in ghost form, Icon is slider and you. can get the svg from the figma, let me know if you face with any issue to take these icon.
2/ Save and cancelbutton is in different design than the figma
3/ the trash icon is also different.

**Comment by xquanluu** [comment]
There is bug that.

I already added 1 account then delete it, then. add the account again then I go back. to Dialer,  the page will show me. Go to. Settings to configure your account

**Comment by xquanluu** [comment]
Screenshot 2024-08-26 at 14 43 42

could you check figma if the bottom footer is inside a grey box.

**Review by xquanluu — CHANGES_REQUESTED** [review]
Please help to check my comments above. let me know if you have something is unclear

**Comment by xquanluu** [comment]
Screenshot 2024-08-27 at 06 31 16
This is figma design,

you can see that all component are aligned, when you open the account in settings, it should expand the existing area belong to the account, instead of open down below area. Here is your version

Screenshot 2024-08-27 at 06 33 28

account item padding is to big compare with figma. please follow figma as much as possible.

When I click on Hoan HL2, the Hoan HL2 border radius rectangle should expand and show detail data instead of open down below area


Account gap seem bigger than figma, please check as well, css is defined in figma, don't use your own css.

**Comment by xquanluu** [comment]
For setting page:
1/ When you click on an account, in figma, we don't open an another area down below the account. the account border radius rectangle itself is an expandable area, when you click on that the heigh of the area is changed and show details of the account. Please ping me on slack if you need my explanation again, check figma again.

2/ There is a bug, if I opened the 1st account, then click on 2nd account, the app will close the detail of 1st account but dont open the 2nd account detail.

Please do open any account if we click on that account and the orther should close automatically.

**Comment by xquanluu** [comment]
Screenshot 2024-08-28 at 05 46 39
Please update Dialer page that if we click out site of account selector, it should close the selector. FOr now, I have to click to selected value to close the selector to make sure the online account is not changed.

**Comment by xquanluu** [comment]
Screenshot 2024-08-30 at 05 14 29

All issue above are good now. Could you please check why the font is not source san 3 if we build the project and install the extension?

**Comment by xquanluu** [comment]
There is another issue:
In dailer tab, if there is active account, and we keep changing account quickly, the app will go crazy that he keep saying cannot connect to the wss.

**Comment by xquanluu** [comment]
Hi @mister-abdurahman 

I'm sorry for wrongly reported issue for changing account quickly, It was not issue.

**Comment by xquanluu** [comment]
Screenshot 2024-09-02 at 15 55 31

Hi @mister-abdurahman 

I'm not able to cancel or drop a call after calling any number, could you please help to check

File: public/index.html [hunk 1]
```diff
@@ -4,6 +4,12 @@
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <!-- <link rel="preconnect" href="https://fonts.googleapis.com" />
```
Comment (xquanluu) [inline public/index.html]: If not used, please remove this

File: src/storage/index.ts [hunk 2]
```diff
@@ -25,21 +27,159 @@ export const deleteConferenceSettings = () => {
...
   const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString(
     "base64"
   );
 
-  localStorage.setItem(SETTINGS_KEY, encoded);
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const parsed = JSON.parse(str);
+
+    // const data: IAppSettings[] = parsed.map((el: saveSettingFormat) => {
```
Comment (xquanluu) [inline src/storage/index.ts]: please clean these commented code

File: src/storage/index.ts [hunk 3]
```diff
@@ -25,21 +27,159 @@ export const deleteConferenceSettings = () => {
...
+      return {
+        active: el.active,
+        decoded: JSON.parse(
+          Buffer.from(el.encoded, "base64").toString("utf-8")
+        ),
+        id: el.id,
+      };
+    });
+    return decoded;
+    // const planText = Buffer.from(str, "base64").toString("utf-8");
```
Comment (xquanluu) [inline src/storage/index.ts]: clean these code please

File: src/storage/index.ts [hunk 4]
```diff
@@ -25,21 +27,159 @@ export const deleteConferenceSettings = () => {
...
+      return {
+        active: el.active,
+        decoded: JSON.parse(
+          Buffer.from(el.encoded, "base64").toString("utf-8")
+        ),
+        id: el.id,
+      };
+    });
+    return decoded.find((el) => el.active) as IAppSettings;
+    // const planText = Buffer.from(str, "base64").toString("utf-8");
```
Comment (xquanluu) [inline src/storage/index.ts]: clean these code

File: src/styles.scss [hunk 5]
```diff
@@ -1,7 +1,31 @@
+@font-face {
+  font-family: "Source Sans";
+  src: url("../public/fonts/SourceSans3-Regular.ttf") format("truetype");
+}
+// @font-face {
```
Comment (xquanluu) [inline src/styles.scss]: remove commented code

**Comment by xquanluu** [comment]
Hi @mister-abdurahman 

outbound call is working but inbound call is not working, could you please help to check this?

File: src/window/footer/footer.tsx [hunk 6]
```diff
@@ -0,0 +1,200 @@
+import { HStack, Image, Text, useToast } from "@chakra-ui/react";
+import jambonz from "src/imgs/jambonz.svg";
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from "react";
+import { SipClientStatus } from "src/common/types";
+import { SipConstants, SipUA } from "src/lib";
+import { DEFAULT_TOAST_DURATION } from "src/common/constants";
+import JambonzSwitch from "src/components/switch";
+import "./styles.scss";
+
+function Footer({
```
Comment (xquanluu) [inline src/window/footer/footer.tsx:8]: Could you please move SIPUA initialization back to phone/index.tsx?

File: src/storage/index.ts (hunk 4 above)
Comment (mister-abdurahman) [reply src/storage/index.ts]: done

File: src/storage/index.ts (hunk 3 above)
Comment (mister-abdurahman) [reply src/storage/index.ts]: done

File: src/storage/index.ts (hunk 2 above)
Comment (mister-abdurahman) [reply src/storage/index.ts]: cleaned all commented code, thanks

**Comment by xquanluu** [comment]
When I activate invalid account and toggle the you are online button
I will get the error message say cannot connect
but the toggle is still saying I'm online
could you please help me to fix this issue as well?

**Comment by mister-abdurahman** [comment]
> When I activate invalid account and toggle the you are online button I will get the error message say cannot connect but the toggle is still saying I'm online could you please help me to fix this issue as well?

yes, that's the issue I called you about, will still check it out though.

**Comment by xquanluu** [comment]
Hi @mister-abdurahman could you please help to resolve conflict

**Comment by mister-abdurahman** [comment]
> Hi @mister-abdurahman could you please help to resolve conflict

Yes, done, you can check it now.

File: src/storage/index.ts [hunk 7]
```diff
@@ -25,21 +27,143 @@ export const deleteConferenceSettings = () => {
...
+}
+
 export const saveSettings = (settings: AppSettings) => {
   const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString(
     "base64"
   );
 
-  localStorage.setItem(SETTINGS_KEY, encoded);
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
```
Comment (xquanluu) [inline src/storage/index.ts]: const str = localStorage.getItem(SETTINGS_KEY);
const parsed = str ? JSON.parse(str) : [];
const newItem = { id: parsed.length + 1, encoded, active: parsed.length === 0 };

localStorage.setItem(SETTINGS_KEY, JSON.stringify([...parsed, newItem]));

Could you rethink to make it simpler like this?

File: src/storage/index.ts [hunk 8]
```diff
@@ -25,21 +27,143 @@ export const deleteConferenceSettings = () => {
...
+
+    localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));
+  }
+};
+export const setActiveSettings = (id: number) => {
   const str = localStorage.getItem(SETTINGS_KEY);
   if (str) {
-    const planText = Buffer.from(str, "base64").toString("utf-8");
-    return JSON.parse(planText) as AppSettings;
+    const parsed = JSON.parse(str);
```
Comment (xquanluu) [inline src/storage/index.ts:80]: const newData = parsed.map((el: saveSettingFormat) => ({
  ...el,
  active: el.id === id,
}));

localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));

maybe this much simpler, could you rethink about this?

File: src/storage/index.ts (hunk 7 above)
Comment (mister-abdurahman) [reply src/storage/index.ts]: okay, on it

File: src/storage/index.ts (hunk 8 above)
Comment (mister-abdurahman) [reply src/storage/index.ts:80]: ok, will do

File: src/storage/index.ts [hunk 9]
```diff
@@ -25,21 +27,143 @@ export const deleteConferenceSettings = () => {
...
+export const getActiveSettings = (): IAppSettings => {
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const data: { active: boolean; encoded: string; id: number }[] =
+      JSON.parse(str);
+    const decoded: IAppSettings[] = data.map((el) => {
+      return {
+        active: el.active,
+        decoded: JSON.parse(
+          Buffer.from(el.encoded, "base64").toString("utf-8")
```
Comment (xquanluu) [inline src/storage/index.ts]: can we find the active one and decode for the active setting, let avoid decode all.

File: src/storage/index.ts [hunk 10]
```diff
@@ -50,16 +174,64 @@ export const saveAddvancedSettings = (settings: AdvancedAppSettings) => {
     "base64"
   );
 
-  localStorage.setItem(ADVANCED_SETTINGS_KET, encoded);
+  const str = localStorage.getItem(ADVANCED_SETTINGS_KET);
```
Comment (xquanluu) [inline src/storage/index.ts:149]: export const saveAddvancedSettings = (settings: AdvancedAppSettings) => {
  const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString("base64");
  const str = localStorage.getItem(ADVANCED_SETTINGS_KET);
  const data = str ? JSON.parse(str) : [];

  // Check if the encoded settings already exist
  if (data.some((el: { encoded: string }) => el.encoded === encoded)) return;

  // Add new settings and save to local storage
  data.push({ encoded, active: data.length === 0, id: data.length + 1 });
  localStorage.setItem(ADVANCED_SETTINGS_KET, JSON.stringify(data));
};


could you rethink to use this code

File: src/storage/index.ts (hunk 9 above)
Comment (mister-abdurahman) [reply src/storage/index.ts]: ✔

File: src/storage/index.ts (hunk 10 above)
Comment (mister-abdurahman) [reply src/storage/index.ts:149]: ✔

**Review by xquanluu — APPROVED** [review]
Look good to me

**Pull request merged**